### PR TITLE
feat(tooltip): add tooltip component

### PR DESCRIPTION
**Pull request by brnrdog**
tooltip

### CODE REVIEW

**Comment by fernandofleury** [comment]
é possível centralizar o caret na tooltip? (centralizar também baseado no lado que ele abre)

**Comment by brnrdog** [comment]
É, está mais para a direita. Vou dar uma olhada nisso, @fernandofleury

**Comment by fernandofleury** [comment]
Se puder revisar o transition-out do tooltip

**Comment by fernandofleury** [comment]
@brnrdog The doc is missing the `jQuery` usage

**Comment by brnrdog** [comment]
done, @fernandofleury!

File: src/css/atoms/tooltip.css [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  visibility: hidden;
+
+  &:parent:hover > .tooltip {
+    opacity: 1;
+    visibility: visible;
+  }
+
+  .tooltip-arrow {
+    border-style: solid;
+    border-width: 7px;
```
Comment (fernandofleury) [inline src/css/atoms/tooltip.css]: Please use rem to refer to units.

File: src/css/atoms/tooltip.css [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  position: absolute;
+  transition: opacity .3s ease-in-out, visibility .6s;
+  visibility: hidden;
+
+  &:parent:hover > .tooltip {
+    opacity: 1;
+    visibility: visible;
+  }
+
+  .tooltip-arrow {
```
Comment (fernandofleury) [inline src/css/atoms/tooltip.css]: I think this guy can move into an atom. And the positioning styles could go into the tooltip molecule

File: src/css/atoms/tooltip.css [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    margin: 7px;
+    position: absolute;
+    width: 0;
+  }
+
+  &[x-placement^="top"] {
+    margin-bottom: 10px;
+  }
+
+  &[x-placement^="top"] .tooltip-arrow {
```
Comment (fernandofleury) [inline src/css/atoms/tooltip.css]: These styles could move into tooltip molecule

File: docs/js/tooltip.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+# Tooltip
+<p class="lead">
+  The tooltip component is used to display a short description about something when the mouse is over a target element.
+</p>
+
+## How to
+
+To use the tooltip component, you need to add `[data-tooltip = 'tooltipSelector']` to the target element (you can use a class name, an id or a data attribute). The selector inserted should be present inside the target element, it'll be the tooltip itself. This component uses the [popper.js](http://popper.js.org/) library to deal with the positioning and jQuery.
```
Comment (fernandofleury) [inline docs/js/tooltip.md]: I think you can spit the jQuery usage into the js usage section

**Comment by fernandofleury** [comment]
@brnrdog Do not forget to include the `dist` folder into the final commit. They are the static files for package usage.

**Comment by leandrogrillo** [comment]
@fernandofleury we already had problems including dist files when merging or rebasing to master.

I think it's better to add these files when create new tag.

File: src/css/atoms/tooltip.css [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  box-shadow: 0 0 3px 0 color($color-default-darker alpha(.3));
+  color: $color-default-dark;
+  font-size: .8rem;
+  opacity: 0;
+  padding: .5rem;
+  position: absolute;
+  transition: opacity .3s ease-in-out, visibility .6s;
+  visibility: hidden;
+
+  &:parent:hover > .tooltip {
```
Comment (leandrogrillo) [inline src/css/atoms/tooltip.css]: is `:parent` a valid selector? 

If it's, which ie version is supported?

Comment (brnrdog) [inline src/css/atoms/tooltip.css]: Actually it isn't, my bad. I'm gonna fix it 😉

**Comment by brnrdog** [comment]
@leandrogrillo should I remove the dist files?

**Comment by fernandofleury** [comment]
@brnrdog  Fine by me

**Comment by leandrogrillo** [comment]
@brnrdog @fernandofleury i prefer to remove the dist files and just add when create a new version of garden

:)

**Comment by brnrdog** [comment]
@leandrogrillo I just deleted the dist files and rebased my branch if that's ok. If you prefer, I can keep the original dist files from the `master` branch or something.

**Comment by leandrogrillo** [comment]
@brnrdog naaah, you don't need to do it. Just remove dist files and that's ok :)

**Comment by leandrogrillo** [comment]
ty for the contribution @brnrdog

**Pull request merged**